lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -44,6 +44,16 @@ module SchedulingExtensions
       # param, otherwise params is passed in as the only parameter to
       # perform.
       def schedule=(schedule_hash)
+        if supports_lua?
+          setup_schedule(schedule_hash)
+        else
+          Resque.logger.warn!('setup_schedule! can cause a race condition, it is ' \
+            'recommended to upgrade your Redis version so that it supports lua SCRIPT')
```

Comment:
The internal method names shouldn't be exposed in this warning. Maybe something like:

"You're running Redis version #{redis_master_version}. Redis version 2.5 is required to run Lua scripts to atomically set up the schedule. `resque-scheduler` will use the pre 2.5 behaviour which can cause a race condition that will result in an empty schedule. It is highly recommended to upgrade Redis for production workloads."